src/main/java/org/spongepowered/common/item/inventory/EmptyInventoryImpl.java

```diff
@@ -225,6 +225,21 @@ public void setMaxStackSize(int size) {
     }
 
     @Override
+    public Inventory intersect(Inventory inventory) {
+        return this; // Shortcut
+    }
+
+    @Override
+    public Inventory union(Inventory inventory) {
+        return inventory; // Shortcut
+    }
+
+    @Override
+    public boolean containsInventory(Inventory inventory) {
+        return this == inventory;
```

Comment:
`return this instanceof EmptyInventoryImpl;`? Or is it intentional to only have the same inventory return `true`?